tensorrt_llm/_torch/pyexecutor/sampler.py

```
@@ -1670,6 +1677,19 @@ def _is_new_request(self, request: LlmRequest) -> bool:
             )
         )
 
+    @override
+    def validate_request(self, request: LlmRequest) -> None:
+        if self._use_beam_search:
+            if request.py_return_log_probs:
+                if request.py_num_logprobs > 1:
+                    raise ValueError(
+                        "Beam search does not support returning multiple logprobs per request"
+                    )
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)